### PR TITLE
[collectd 6] disk: migration to v6.0

### PR DESCRIPTION
**Pull request by manuelluis**
ChangeLog: disk plugin: migration to v6.0

### CODE REVIEW

File: src/disk.c [hunk 1]
```diff
@@ -560,12 +568,30 @@ static int disk_read(void) {
...
-      disk_submit(disk_name, "disk_octets", read_byt, write_byt);
-    if ((read_ops != -1LL) || (write_ops != -1LL))
-      disk_submit(disk_name, "disk_ops", read_ops, write_ops);
-    if ((read_tme != -1LL) || (write_tme != -1LL))
-      disk_submit(disk_name, "disk_time", read_tme / 1000, write_tme / 1000);
+    metric_t m = {0};
+    metric_label_set(&m, "device", disk_name);
+    if ((read_byt != -1LL) || (write_byt != -1LL)) {
+      m.value.counter = read_byt;
+      metric_family_metric_append(&fam_isk_read_bytes, m);
```
Comment (kwiatrox) [inline src/disk.c:575]: fam_disk_read_bytes

File: src/disk.c [hunk 2]
```diff
@@ -660,36 +686,53 @@ static int disk_read(void) {
...
     }
     if (devstat_compute_statistics(snap_iter, NULL, 1.0, DSM_TOTAL_BUSY_TIME,
                                    &busy_time, DSM_TOTAL_DURATION,
                                    &total_duration, DSM_QUEUE_LENGTH,
                                    &queue_length, DSM_NONE) != 0) {
       WARNING("%s", devstat_errbuf);
     } else {
-      submit_io_time(disk_name, busy_time, total_duration);
-      submit_in_progress(disk_name, (gauge_t)queue_length);
+      m.value.counter = busy_time;
```
Comment (kwiatrox) [inline src/disk.c:726]: Cast from `double` to `counter_t`, the same as for `write_time` and `read_time`.

File: src/disk.c [hunk 3]
```diff
@@ -660,36 +686,53 @@ static int disk_read(void) {
...
+      metric_family_metric_append(&fam_disk_io_time, m);
+
+      m.value.counter = total_duration;
```
Comment (kwiatrox) [inline src/disk.c:729]: Cast to `counter_t`?

**Review by kwiatrox — COMMENTED** [review]
Thank you @manuelluis for your contribution.
It seems that it was not tested on macOS wih IOKit, I can't do it now, but probably at least compilation should be checked with IOKit lib.

File: src/disk.c (hunk 1 above)
Comment (manuelluis) [reply src/disk.c]: Fixed.

File: src/disk.c [hunk 4]
```diff
@@ -560,12 +568,30 @@ static int disk_read(void) {
...
+    if ((read_byt != -1LL) || (write_byt != -1LL)) {
+      m.value.counter = read_byt;
+      metric_family_metric_append(&fam_isk_read_bytes, m);
+
+      m.value.counter = write_byt;
+      metric_family_metric_append(&fam_disk_write_bytes, m);
+    }
+    if ((read_ops != -1LL) || (write_ops != -1LL)) {
+      m.value.counter =
+          read_ops metric_family_metric_append(&fam_disk_read_ops, m);
```
[the remark this answers is not in the document]
Comment (manuelluis) [reply src/disk.c]: Fixed

File: src/disk.c [hunk 5]
```diff
@@ -560,12 +568,30 @@ static int disk_read(void) {
...
+
+      m.value.counter =
+          write_ops metric_family_metric_append(&fam_disk_write_ops, m);
```
[the remark this answers is not in the document]
Comment (manuelluis) [reply src/disk.c]: Fixed.

File: src/disk.c (hunk 2 above)
Comment (manuelluis) [reply src/disk.c]: Casting added.

File: src/disk.c (hunk 3 above)
Comment (manuelluis) [reply src/disk.c]: Casting added.

**Comment by mrunge** [comment]
This was ported by @hnez in https://github.com/collectd/collectd/pull/4075

**Pull request closed without merging**